WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsDialogFragment.kt

```diff
@@ -0,0 +1,399 @@
+        mOptionsView?.orientation = LinearLayout.VERTICAL
+        innerView.addView(mDisabledView)
+        innerView.addView(mOptionsView)
+        outerView.addView(innerView)
+        configureLayoutForView(mOptionsView!!)
+
+        val builder: AlertDialog.Builder = MaterialAlertDialogBuilder(requireActivity()).apply {
+            setTitle(title)
+            setPositiveButton(android.R.string.ok, this@NotificationsSettingsDialogFragment)
+            setNegativeButton(R.string.cancel, this@NotificationsSettingsDialogFragment)
+            setView(layout)
+        }
+
+        if (mShouldDisplayMainSwitch) {
+            setupTitleViewWithMainSwitch(outerView)
+            if (mTitleViewWithMainSwitch == null)
+                AppLog.e(AppLog.T.NOTIFS, "Main switch enabled but layout not set")
+            else
+                builder.setCustomTitle(mTitleViewWithMainSwitch)
+        }
+
+        return builder.create()
+    }
+
+    private fun setupTitleViewWithMainSwitch(view: View) {
+        when (this.channel) {
+            NotificationsSettings.Channel.BLOGS -> if (this.type == NotificationsSettings.Type.TIMELINE) {
+                mTitleViewWithMainSwitch = layoutInflater
+                    .inflate(R.layout.notifications_tab_for_blog_title_layout, view as ViewGroup, false) as ViewGroup
+            }
+
+            NotificationsSettings.Channel.OTHER, NotificationsSettings.Channel.WPCOM -> {}
+        }
+        if (mTitleViewWithMainSwitch != null) {
+            val titleView = mTitleViewWithMainSwitch!!.findViewById<TextView>(R.id.title)
+            titleView.text = title
+            mMainSwitchToolbarView = mTitleViewWithMainSwitch!!.findViewById(R.id.main_switch)
+            mMainSwitchToolbarView!!.setMainSwitchToolbarListener(this)
+            mMainSwitchToolbarView!!
+                .setBackgroundColor(ContextCompat.getColor(requireContext(), android.R.color.transparent))
+
+            // Main Switch initial state:
+            // On: If at least one of the settings options is on
+            // Off: If all settings options are off
+            val settingsJson = settings.getSettingsJsonForChannelAndType(channel, type, blogId)
+            val checkMainSwitch = settings.isAtLeastOneSettingsEnabled(
+                settingsJson,
+                mSettingsArray,
+                mSettingsValues
+            )
+            mMainSwitchToolbarView!!.loadInitialState(checkMainSwitch)
+            hideDisabledView(mMainSwitchToolbarView!!.isMainChecked)
+        }
+    }
+
+    override fun onClick(dialog: DialogInterface?, which: Int) {
+        if (which == DialogInterface.BUTTON_POSITIVE && mUpdatedJson.length() > 0) {
+            onNotificationsSettingsChangedListener.onSettingsChanged(this.channel,
+                this.type, this.blogId, mUpdatedJson)
+
+            // Update the settings json
+            val keys: Iterator<*> = mUpdatedJson.keys()
+            while (keys.hasNext()) {
```

Comment:
What would you say of simplifying the `while` block with something like the following:
```
for (settingName in mUpdatedJson.keys()) {
                settings.updateSettingForChannelAndType(
                    this.channel, this.type, settingName,
                    mUpdatedJson.optBoolean(settingName), this.blogId
                )
            }
```